src/components/Download/DownloadOptionsForm/index.js

```diff
@@ -44,9 +47,10 @@ export const DownloadOptionsForm = ({
       const response = await onSubmit(downloadOptions)
       pathname = response
     } else {
-      await updateDataset(selectedDataset.id, {
+      const datasetToUpdate = isProcessed ? regenratedDataset : selectedDataset
+      await updateDataset(datasetToUpdate.id || datasetToUpdate.id, {
```

Comment:
Was this intentional? Looks like you might have expected a different fallback.
```suggestion
      await updateDataset(datasetToUpdate.id, {
```

Reply:
Oh, thank you for catching that (8da4d1aa8015cf1f7f6a410631bc5f7a004dabb8)!